schema-serializer/src/main/java/io/confluent/kafka/formatter/SchemaMessageReader.java

```diff
@@ -153,6 +150,38 @@ protected ParsedSchema parseSchema(
     return provider.parseSchema(schema, references).get();
   }
 
+  private ParsedSchema getSchema(SchemaRegistryClient schemaRegistry,
+                                 Properties props,
+                                 boolean isKey) {
+    ParsedSchema schema = getSchemaById(schemaRegistry, props, isKey);
+    if (schema != null) {
+      return schema;
+    }
+    String schemaString = getSchemaString(props, isKey);
+    List<SchemaReference> refs = getSchemaReferences(props, isKey);
+    return parseSchema(schemaRegistry, schemaString, refs);
+  }
+
+  private ParsedSchema getSchemaById(SchemaRegistryClient schemaRegistry,
```

Comment:
IDK about naming on this. 

How about
```suggestion
  private ParsedSchema getSchemaId(SchemaRegistryClient schemaRegistry,
```

Reply:
Thanks @cricket007 , it's short for "get ParsedSchema by its ID", rather than "get the ParsedSchema's ID"